scalyr_agent/copying_manager.py

```diff
@@ -224,6 +226,32 @@ def __init__(self, configuration, monitors):
         StoppableThread.__init__(self, name="log copier thread")
         self.__config = configuration
 
+        self._session = None
+        self._control_plane_client = None
+        self._data_plane_client = None
+        if self.__config.use_new_ingestion:
+            from scalyr_ingestion_client.session import (  # pylint: disable=import-error
+                Session,
+            )
+            from scalyr_ingestion_client.client import (  # pylint: disable=import-error
+                ControlPlaneAPIClient,
+                DataPlaneAPIClient,
+            )
+
+            self._session = Session(uuid=str(uuid.uuid4()))
+
+            self._control_plane_client = ControlPlaneAPIClient(
```

Comment:
While testing those changes, I noticed ``send_hello`` gets called every time we send events.

> 3:21:00.722 110 I requestDeserializationNanos=60022 tag=ingestionManager.clientHello 
3:21:00.722 111 I requestDeserializationNanos=188475 tag=ingestionManager.sendEvents 
3:21:00.727 115 I requestDeserializationNanos=189423 tag=ingestionManager.sendEvents 
3:21:01.280 46 I path=/home/kami/w/scalyr/ingestion-pipeline/.scalyrDbState/env/eventPersistence/directdb/meta debugName=DirectDB freeSpace=388148609024 tag=getFreeSpace usableSpace=343632363520 value=343632363520 
3:21:01.321 52 I source=counters tag=directDbAllMemtableMemory_hires value=0 
3:21:01.336 51 I path=/home/kami/w/scalyr/ingestion-pipeline/.scalyrDbState/env/eventPersistence/directdb debugName=DirectDB freeSpace=388148609024 tag=getFreeSpace usableSpace=343632363520 value=343632363520 
3:21:05.734 118 I requestDeserializationNanos=166601 tag=ingestionManager.sendEvents 
3:21:05.747 122 I requestDeserializationNanos=156924 tag=ingestionManager.sendEvents 
3:21:05.753 123 I requestDeserializationNanos=2172218 tag=ingestionManager.clientHello 
3:21:06.153 58 I node=localhost service=INGESTION_MANAGER action=heartbeat tag=serviceDiscovery isDrained=false isHealthy=true 
3:21:06.321 52 I source=counters tag=directDbAllMemtableMemory_hires value=0 
3:21:10.836 126 I requestDeserializationNanos=176803 tag=ingestionManager.clientHello 
3:21:11.284 46 I path=/home/kami/w/scalyr/ingestion-pipeline/.scalyrDbState/env/eventPersistence/directdb/meta debugName=DirectDB freeSpace=388148506624 tag=getFreeSpace usableSpace=343632261120 value=343632261120 
3:21:11.322 52 I source=counters tag=directDbAllMemtableMemory_hires value=0 
3:21:11.338 51 I path=/home/kami/w/scalyr/ingestion-pipeline/.scalyrDbState/env/eventPersistence/directdb debugName=DirectDB freeSpace=388148506624 tag=getFreeSpace usableSpace=343632261120 value=343632261120 
3:21:11.354 53 I tag=cpuPing 
3:21:11.838 49 I tag=compactionScannerStatus position=[[empty]] recompressScanPosition=[[empty]] level=2 
3:21:15.721 58 I node=localhost service=INGESTION_MANAGER action=heartbeat tag=serviceDiscovery isDrained=false isHealthy=true 
3:21:15.756 129 I requestDeserializationNanos=604205 tag=ingestionManager.sendEvents 
3:21:15.864 132 I requestDeserializationNanos=105244 tag=ingestionManager.clientHello 
3:21:16.323 52 I source=counters tag=directDbAllMemtableMemory_hires value=0 
3:21:20.773 135 I requestDeserializationNanos=686218 tag=ingestionManager.sendEvents 
3:21:20.885 138 I requestDeserializationNanos=91936 tag=ingestionManager.clientHello 
3:21:21.286 46 I path=/home/kami/w/scalyr/ingestion-pipeline/.scalyrDbState/env/eventPersistence/directdb/meta debugName=DirectDB freeSpace=388148428800 tag=getFreeSpace usableSpace=343632183296 value=343632183296 
3:21:21.322 52 I source=counters tag=directDbAllMemtableMemory_hires value=0 

This needs to be refactored so we only call it once on start up - similar to when and how we do remote server check with the current API.